addons/uom/models/uom_uom.py

```diff
@@ -12,11 +42,17 @@ class UoMCategory(models.Model):
     name = fields.Char('Unit of Measure Category', required=True, translate=True)
 
     def unlink(self):
-        uom_categ_unit = self.env.ref('uom.product_uom_categ_unit')
-        uom_categ_wtime = self.env.ref('uom.uom_categ_wtime')
-        if any(categ.id in (uom_categ_unit + uom_categ_wtime).ids for categ in self):
-            raise UserError(_("You cannot delete this UoM Category as it is used by the system."))
-        return super(UoMCategory, self).unlink()
+        linked_model_data = self.env['ir.model.data'].search([
+            ('model', '=', self._name),
+            ('res_id', 'in', self.ids),
+        ])
+        if linked_model_data:
+            locked_categories = self.browse(linked_model_data.mapped('res_id'))
+            raise UserError(_(
+                "The following categories of unit of measure are protected and cannot be modified: %s",
+                ", ".join(locked_categories.mapped('name')),
+            ))
+        return super().unlink()
```

Comment:
idea: could we just replace this whole thing with a simple `ondelete='restrict'` on [categ_id](https://github.com/odoo/odoo/blob/master/addons/uom/models/uom_uom.py#L29)?